docs/examples/tutorial/string/c_func.py

```diff
@@ -0,0 +1,31 @@
+from cython.cimports.libc.stdlib import malloc
+from cython.cimports.libc.string import strcpy, strlen
+
+hello_world = cython.declare(cython.p_char, 'hello world')
+n = cython.declare(cython.Py_ssize_t, strlen(hello_world))
+
+
+# this function is defined as a cdef function in the .pxd file, no need for @cython.cfunc
+def c_call_returning_a_c_string() -> cython.p_char:
+    c_string: cython.p_char = cython.cast(cython.p_char, malloc(
+        (n + 1) * cython.sizeof(cython.char)))
+
+    if not c_string:
+        return cython.NULL  # malloc failed
+
+    strcpy(c_string, hello_world)
+    return c_string
+
+
+# this function is defined as a cdef function in the .pxd file, no need for @cython.cfunc
+@cython.cfunc  # DELETE ME (GH#5970)
```

Comment:
This is a blocker

Reply:
See fix #5974

Reply:
If I understand correctly, the example is working even without https://github.com/cython/cython/pull/5974. It is more nice to have it fixed right?

Reply:
Yes, it works in the way it's currently implemented but it should work without cfunc, like in all other examples, which it doesn't.
The fix appears not that big to postpone it's merge, so it would probably be better to merge it, fix it here and merge this.